src/utils/daos_hdlr.c

```diff
@@ -1535,6 +1536,741 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	char *cont_name = NULL;
+	int rc = 0;
+	if (path == NULL || _obj_name == NULL || _cont_name == NULL)
+			return -EINVAL;
+
+	if (strcmp(path, "/") == 0) {
+		*_cont_name = strdup("/");
+		if (*_cont_name == NULL)
+			return -ENOMEM;
+		*_obj_name = NULL;
+		return 0;
+	}
+	f1 = strdup(path);
+	if (f1 == NULL) {
+		rc = -ENOMEM;
+		goto out;
+	}
+
+	f2 = strdup(path);
+	if (f2 == NULL) {
+		rc = -ENOMEM;
+		goto out;
+	}
+	fname = basename(f1);
+	cont_name = dirname(f2);
+
+	if (cont_name[0] == '.' || cont_name[0] != '/') {
+		char cwd[1024];
+
+		if (getcwd(cwd, 1024) == NULL) {
+			rc = -ENOMEM;
+			goto out;
+		}
+
+		if (strcmp(cont_name, ".") == 0) {
+			cont_name = strdup(cwd);
+			if (cont_name == NULL) {
+				rc = -ENOMEM;
+				goto out;
+			}
+		} else {
+			char *new_dir = calloc(strlen(cwd) + strlen(cont_name) + 1, sizeof(char));
+			if (new_dir == NULL) {
+				rc = -ENOMEM;
+				goto out;
+			}
+
+			strcpy(new_dir, cwd);
+			if (cont_name[0] == '.') {
+				strcat(new_dir, &cont_name[1]);
+			} else {
+				strcat(new_dir, "/");
+				strcat(new_dir, cont_name);
+			}
+			cont_name = new_dir;
+		}
+		*_cont_name = cont_name;
+	} else {
+		*_cont_name = strdup(cont_name);
+		if (*_cont_name == NULL) {
+			rc = -ENOMEM;
+			goto out;
+		}
+	}
+	*_obj_name = strdup(fname);
+	if (*_obj_name == NULL) {
+		free(*_cont_name);
+		*_cont_name = NULL;
+		rc = -ENOMEM;
+		goto out;
+	}
+	out:
+		if (f1)
+			free(f1);
+		if (f2)
+			free(f2);
+	return rc;
+}
+
+static ssize_t
+daos_write(struct daos_file_t *daos_file, char *file, void *buf, size_t size)
+{
+	int rc;
+	/* record address and size of user buffer in io vector */
+	d_iov_t iov;
+	d_iov_set(&iov, buf, size);
+
+	/* define scatter-gather list for dfs_write */
+	d_sg_list_t sgl;
+	sgl.sg_nr = 1;
+	sgl.sg_iovs = &iov;
+	sgl.sg_nr_out = 1;
+
+	rc = dfs_write(daos_file->dfs, daos_file->obj, &sgl, daos_file->offset, NULL); 
+
+	if (rc) {
+		fprintf(stderr, "dfs_write %s failed (%d %s)",
+			file, rc, strerror(rc));
+		errno = rc;
+		size = -1;
+	}
+
+	daos_file->offset += (daos_off_t)size;
+	return (ssize_t) size;
+}
+
+static ssize_t
+daos_file_write(daos_file_t *daos_file, char* file,
+		void* buf, size_t size)
+{
+	ssize_t num_bytes_written = 0;
+	if (daos_file->type == POSIX) {
+		num_bytes_written = write(daos_file->fd, buf, size);
+	} else if (daos_file->type == DAOS) {
+		num_bytes_written = daos_write(daos_file, file, buf, size);
+	} else {
+		fprintf(stderr, "File type not known: %s type=%d",
+			file, daos_file->type);
+	}
+	if (num_bytes_written < 0) {
+		fprintf(stderr, "write error on %s type=%d\n",
+			file, daos_file->type);
+	}
+	return num_bytes_written;
+}
+
+static void 
+daos_open(daos_file_t *daos_file, char *file, int flags, mode_t mode)
+{
+
+	int rc; 
+	dfs_obj_t *parent = NULL;
+	char *name = NULL, *dir_name = NULL;
+
+	parse_filename(file, &name, &dir_name);
+
+	assert(dir_name);
+
+	rc = dfs_lookup(daos_file->dfs, dir_name, O_RDWR, &parent, NULL, NULL);
```

Comment:
In a future iteration, we might consider using a `d_hash` to cache some lookups

Reply:
@daltonbohning yep it would be nice if we can get to where we can just re-use the I/O wrappers across DAOS and mpifileutils

Reply:
i think performance wise, it's pretty important to add a hashtable there. 
i've done something similar in the ior DFS driver.